cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -597,15 +597,16 @@ func ValidateAPIEndpoint(c *kubeadm.APIEndpoint, fldPath *field.Path) field.Erro
 // ValidateIgnorePreflightErrors validates duplicates in:
 // - ignore-preflight-errors flag and
 // - ignorePreflightErrors field in {Init,Join}Configuration files.
+// and make sure ignore "all" cannot configured with individual checks.
 func ValidateIgnorePreflightErrors(ignorePreflightErrorsFromCLI, ignorePreflightErrorsFromConfigFile []string) (sets.Set[string], error) {
 	ignoreErrors := sets.New[string]()
 	allErrs := field.ErrorList{}
+	ignorePreflightErr := ignorePreflightErrorsFromConfigFile
 
-	for _, item := range ignorePreflightErrorsFromConfigFile {
-		ignoreErrors.Insert(strings.ToLower(item)) // parameters are case insensitive
+	if ignorePreflightErrorsFromCLI != nil {
```

Comment:
/hold
This seems a break change, we should be careful.
kubeadm will merge all the ignoreErrors from CLI and config file, and use `set` to remove duplicates, that doesn't seem to be a problem. 
cc @neolit123 @pacoxu

Reply:
i think we discussed somewhere, recently (sorry my memory is short) that flags should override config.
if we are merging slices it becomes confusing, but overall, i agree there isn't a huge issue about this.
we are just inconsistent.

with the CLI being GA we cannot introduce any breaking changes, so i wonder what can be done here.
maybe use a feature gate? for example:

- add the FG `MergeIgnorePreflightErrors` with a default value of `false` in 1.29
this changes the behavior but allows opt-out via the FG
- remove the FG in 1.30

this gives users one release to adapt. maybe such a plan will be OK.

@chendave do you know if we have any other places where we are inconsistent - e.g. merging vs override?
what about feature gates, extra args, extra sans from CLI, etc?

Reply:
OK, I'm +1 for adding a new FG in the migration stage.

Reply:
> do you know if we have any other places where we are inconsistent - e.g. merging vs override?

this is the only case of merging instead of override to the best of my knowledge, I am in favor of a more generic name though.

+1 too for a feature gate.

Reply:
Good idea. A FG helps in this case.

Reply:
Done

Reply:
> * add the FG `MergeIgnorePreflightErrors` with a default value of `false` in 1.29
>   this changes the behavior but allows opt-out via the FG

In the current PR, the `FG=true` only merged ignore preflights with config.

> if there are more instances we can call such a FG more generically: `MergeCLISliceArgumentsWithConfig` (just a name idea)

All other flags will override the config directly. So here we use `MergeCLIArgumentsWithConfig`. Did I understand correctly?

Reply:
Right, `ignorePreflightErrors` is the only exception that is merging of (config and flags) instead of override.

Reply:
should we document it somewhere? Or with the feature gate  https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/#feature-gates.

Reply:
will do, thanks for the link!

Reply:
@pacoxu pls see: https://github.com/kubernetes/website/pull/43510